x-pack/plugins/uptime/public/components/functional/charts/duration_chart.tsx

```diff
@@ -0,0 +1,148 @@
+  meanColor,
+  rangeColor,
+}: DurationChartProps) => {
+  const { absoluteStartDate, absoluteEndDate } = useContext(UptimeSettingsContext);
+  // this id is used for the area chart representing the max/min of check durations
+  const areaSpecId = getSpecId('area');
+
+  // this id is used for the line chart representing the average duration length
+  const averageSpecId = getSpecId('average');
+
+  return (
+    <React.Fragment>
+      <EuiTitle size="xs">
+        <h4>
+          <FormattedMessage
+            id="xpack.uptime.monitorCharts.monitorDuration.titleLabel"
+            defaultMessage="Monitor duration"
+            description="The 'ms' is an abbreviation for milliseconds."
+          />
+        </h4>
+      </EuiTitle>
+      <EuiPanel>
+        <Chart className={'story-chart'}>
+          <Axis
+            id={getAxisId('bottom')}
+            position={Position.Bottom}
+            showOverlappingTicks={true}
+            tickFormat={timeFormatter(getChartDateLabel(absoluteStartDate, absoluteEndDate))}
+            title={i18n.translate('xpack.uptime.monitorCharts.durationChart.bottomAxis.title', {
+              defaultMessage: 'Timestamp',
+            })}
+          />
+          <Axis
+            domain={{ min: 0 }}
+            id={getAxisId('left')}
+            position={Position.Left}
+            tickFormat={d => Number(d).toFixed(0)}
+            title={i18n.translate('xpack.uptime.monitorCharts.durationChart.leftAxis.title', {
+              defaultMessage: 'Duration ms',
+            })}
+          />
+          <AreaSeries
+            curve={CurveType.CURVE_MONOTONE_X}
+            customSeriesColors={getColorsMap(rangeColor, areaSpecId)}
+            data={durationArea.map(({ x, yMin, yMax }) => ({
+              x,
+              Min: microsToMillis(yMin),
+              Max: microsToMillis(yMax),
+            }))}
+            id={areaSpecId}
+            name={i18n.translate(
+              'xpack.uptime.monitorCharts.monitorDuration.series.durationRangeLabel',
+              {
+                defaultMessage: 'Duration range',
+              }
+            )}
+            xAccessor={'x'}
+            xScaleType={ScaleType.Time}
+            yAccessors={['Max']}
+            yScaleType={ScaleType.Linear}
+            y0Accessors={['Min']}
+            yScaleToDataExtent={false}
+          />
+          <LineSeries
+            curve={CurveType.CURVE_MONOTONE_X}
+            customSeriesColors={getColorsMap(meanColor, averageSpecId)}
+            data={durationLine.map(({ x, y }) => [x || 0, microsToMillis(y)])}
+            id={averageSpecId}
+            name={i18n.translate(
+              'xpack.uptime.monitorCharts.monitorDuration.series.meanDurationLabel',
+              {
+                defaultMessage: 'Mean duration',
+              }
+            )}
+            xAccessor={0}
+            xScaleType={ScaleType.Time}
+            yAccessors={[1]}
+            yScaleToDataExtent={true}
```

Comment:
The area series has `yScaleToDataExtent={false}` and line has `yScaleToDataExtent={true}`, in this case you are forcing `yScaleToDataExtent = true` for all series. this means that the area series will be drawn with the extent of the lineseries. Is that the desired behaviour?

Reply:
Resolved by https://github.com/elastic/kibana/pull/34313/commits/ab546982ac90d84b40e6978480f882bb0ea09411.
I just set the prop to `false`.